subsys/net/l2_wifi_if_conn/l2_wifi_conn.c

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/net/net_if.h>
+#include <zephyr/net/wifi_mgmt.h>
+#include <zephyr/net/net_event.h>
+#include <net/wifi_mgmt_ext.h>
+#include <net/l2_wifi_connect.h>
+#include <zephyr/net/net_l2_connectivity.h>
+
+int net_l2_wifi_connect(const struct net_if_conn *if_conn)
+{
+	if (!if_conn) {
+		return -EINVAL;
+	}
+
+	int rc = net_mgmt(NET_REQUEST_WIFI_CONNECT_STORED, if_conn->iface, NULL, 0);
+
+	return rc;
```

Comment:
This seems reasonable

How are persistence and timeout handled?

Reply:
We do have a connect timeout that can be configured through net_mgmt, once the timeout is reached connection is cancelled to the WPA supplicant.

Reply:
The connectivity API you are binding via `NET_L2_CONNECTIVITY_DEFINE` defines two configurable parameters:

1. Connectivity persistence
2. Timeout

An L2 connectivity implementation is not complete without providing the behaviors associated with these configurable parameters.

> We dont have timeout for a connection request. Supplicant keeps trying to associate with some backoff interval between each attempt.

The "supplicant keeps trying to associate" part is persistence. If persistence is enabled, it should do exactly as you have described. If persistence is disabled, it should try only once to connect.

Timeout, in this case, would be the amount of time to spend continually re-trying until giving up on the initial connection request.

If the underlying supplicant cannot support these features, you will need to write a wrapper thread that provides them

Reply:
@glarsennordic , I have completed the changes for timeout and persistence. Please have a look and let me know if anything needs to be changed. I am seeing a minor issue regarding with checking for connection status after issuing NET_REQUEST_WIFI_CONNECT_STORED, which I am checking now. Will take this PR out of draft once that is resolved.